### PR TITLE
Feature/leaf 2627/advanced option display

### PR DESCRIPTION
**Pull request by aerinkayne**
Adds a property used to conditionally render a folder icon instead of a notepad icon for the 'edit indicator' button if the indicator has Advanced Options in order to more easily identify indicators that have additional code in the Form Editor UI.

### CODE REVIEW

**Review by Pelentan — APPROVED** [review]

**Review by AmandaSumner — APPROVED** [review]

**Review by gerald-todd — APPROVED** [review]

**Review by raphaelfretter — APPROVED** [review]

**Review by Pelentan — APPROVED** [review]

File: LEAF_Request_Portal/form.php [hunk 1]
```diff
@@ -495,6 +495,8 @@ public function getIndicator($indicatorID, $series, $recordID = null, $parseTemp
         $form[$idx]['isWritable'] = $this->hasWriteAccess($recordID, $data[0]['categoryID']);
         $form[$idx]['isMasked'] = isset($data[0]['groupID']) ? $this->isMasked($data[0]['indicatorID'], $recordID) : 0;
         $form[$idx]['sort'] = $data[0]['sort'];
+        $form[$idx]['has_code'] = ($data[0]['html'] != null && trim($data[0]['html']) != '') ||
```
Comment (gerald-todd) [inline LEAF_Request_Portal/form.php:498]: Checking if `$data[0]['groupID'])` is not null is redundant here; `trim($var)` returns the empty string when `$var` is null.

File: LEAF_Request_Portal/form.php [hunk 2]
```diff
@@ -3513,6 +3515,8 @@ private function buildFormTree($id, $series = null, $recordID = null, $parseTemp
                 $child[$idx]['isWritable'] = $this->hasWriteAccess($recordID, $field['categoryID']);
                 $child[$idx]['isMasked'] = isset($data[$idx]['groupID']) ? $this->isMasked($field['indicatorID'], $recordID) : 0;
                 $child[$idx]['sort'] = $field['sort'];
+                $child[$idx]['has_code'] = ($field['html'] != null && trim($field['html']) != '') ||
```
Comment (gerald-todd) [inline LEAF_Request_Portal/form.php:3518]: Same as above

**Review by maxrichardjr — APPROVED** [review]
See Gerald's Comments about using trim()

File: LEAF_Request_Portal/form.php (hunk 1 above)
Comment (aerinkayne) [reply LEAF_Request_Portal/form.php]: I removed the null checks

**Review by gsteinhoff — APPROVED** [review]

**Pull request merged**